### PR TITLE
Migrate filewidget to functional component

### PR DESCRIPTION
**Pull request by jimmycallin**
### Reasons for making this change

Migrated FileWidget to functional component.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by jimmycallin** [comment]
I'm honestly quite confused about why the test breaks. Would be happy to get some support to figure it out!

**Comment by heath-freenome** [comment]
@jimmycallin I'm on vacation until 8/17 so I won't get a chance to look until after then... sorry

**Comment by jimmycallin** [comment]
no worries! i'll close these two PRs and can open them again once we've figured out the build issues

**Comment by heath-freenome** [comment]
@jimmycallin I think I might have a work-around for this... in each theme (including core), update (or add if missing) the `jest.config.js` to add the following:
```
  "moduleNameMapper": {
    "react": "<rootDir>/node_modules/react"
    "react-dom": "<rootDir>/node_modules/react-dom"
  },
```
and try running the tests again. If that doesn't work try:
```
  "moduleNameMapper": {
    "react": "<rootDir>/node_modules/react/index.js"
    "react-dom": "<rootDir>/node_modules/react-dom/index.js"
  },
```

**Comment by nickgros** [comment]
I got the antd tests to pass with this in packages/antd/jest.config.js

```js
module.exports = {
    moduleNameMapper: {
        "^react$": "<rootDir>/node_modules/react",
        "^react-dom$": "<rootDir>/node_modules/react-dom",
      },
  };
```

**Comment by heath-freenome** [comment]
@jimmycallin If https://github.com/rjsf-team/react-jsonschema-form/pull/3030 passes then, once you rebase on top of the merged version of that PR, yours should pass as well

**Comment by jimmycallin** [comment]
> @jimmycallin If #3030 passes then, once you rebase on top of the merged version of that PR, yours should pass as well

nicely done! i'll rebase these PRs as soon as that is merged

**Comment by heath-freenome** [comment]
@jimmycallin rebase away

File: packages/core/src/components/widgets/FileWidget.tsx [hunk 1]
```diff
@@ -81,81 +81,63 @@ function extractFileInfo(dataURLs: string[]) {
...
-    if (!event.target.files) {
-      return;
-    }
-    processFiles(event.target.files).then((filesInfo) => {
-      const state = {
-        values: filesInfo.map((fileInfo) => fileInfo.dataURL),
-        filesInfo,
-      };
-      this.setState(state, () => {
+  const handleChange = useCallback(
```
Comment (heath-freenome) [inline packages/core/src/components/widgets/FileWidget.tsx:106]: Hey Jimmy, feel like making a pass through the fields and widgets to add `useCallback` in all the right places after this PR merges?

**Review by heath-freenome — APPROVED** [review]

File: packages/core/src/components/widgets/FileWidget.tsx (hunk 1 above)
Comment (jimmycallin) [reply packages/core/src/components/widgets/FileWidget.tsx:106]: sure thing!

**Pull request merged**